nikola/utils.py

```diff
@@ -1938,19 +1943,19 @@ def sort_posts(posts, *keys):
     return posts
 
 
-def smartjoin(join_char: str, string_or_iterable) -> str:
-    """Join string_or_iterable with join_char if it is iterable; otherwise converts it to string.
+def smartjoin(join_char: str, string_or_iterable: Union[None, str, bytes, Iterable[Any]]) -> str:
+    """Join string_or_iterable with join_char if it is iterable; otherwise convert it to string.
 
     >>> smartjoin('; ', 'foo, bar')
     'foo, bar'
     >>> smartjoin('; ', ['foo', 'bar'])
     'foo; bar'
     >>> smartjoin(' to ', ['count', 42])
     'count to 42'
+
+    The treatment of bytes (calling str(string_or_iterable)) is somewhat dubious. Is this needed?
     """
-    if isinstance(string_or_iterable, (str, bytes)):
-        return string_or_iterable
-    elif isinstance(string_or_iterable, Iterable):
+    if isinstance(string_or_iterable, Iterable):
```

Comment:
This won’t work, please undo:

```pycon
>>> isinstance("a", typing.Iterable)
True
```

Reply:
Good point, thanks!

But what we had previously also doesn't work, as `bytes` isn't a `str`.

How about calling `str( )` for both `str` (where it is a no-op) and `bytes` (where you can be lucky, so it might work)?